### PR TITLE
#117: Added exasol-testcontainers for integration tests

### LINKED ISSUE #117: Add exasol-testcontainers for integration tests
### Acceptance Criteria

- Added parquet tests
- Added orc and avro tests

### LINKED ISSUE #118: Support Docker Installations
## Situation

The importing of columnar format does not work from the Exasol Docker installations because of the missing Hadoop user. We should create UDF scriptclasses dedicated for docker setups.

## Acceptance Criteria

- UDF scriptclasses are added
- User guide is updated with deployment setup for docker

### PR DESCRIPTION
**Pull request by morazow**
Fixes #117
Fixes #118

### CODE REVIEW

**Review by AnastasiiaSergienko — COMMENTED** [review]
Checked this and have a few questions :)

File: src/main/scala/com/exasol/cloudetl/scriptclasses/DockerFilesMetadataReader.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * A files metadata reader class to run inside the Exasol docker container.
+ */
+object DockerFilesMetadataReader {
+
+  def run(metadata: ExaMetadata, iterator: ExaIterator): Unit = {
+    import org.apache.hadoop.security.UserGroupInformation
+    UserGroupInformation.setLoginUser(UserGroupInformation.createRemoteUser("exauser"))
+    ImportMetadata.run(metadata, iterator)
```
[the remark this answers is not in the document]
Comment (morazow) [reply src/main/scala/com/exasol/cloudetl/scriptclasses/DockerFilesMetadataReader.scala:14]: Yes. I am going to change others also before the release in a separate ticket & pull request.

File: src/it/scala/com/exasol/cloudetl/DataImporterIT.scala [hunk 2]
```diff
@@ -0,0 +1,1108 @@
...
+    conf = new Configuration
+  }
+
+  override def nestedSuites = Vector(
+    new AvroDataImporter(),
+    new OrcDataImporter(),
+    new ParquetDataImporter()
+  )
+
+  class AvroDataImporter extends BaseDataImporter {
```
[the remark this answers is not in the document]
Comment (morazow) [reply src/it/scala/com/exasol/cloudetl/DataImporterIT.scala:68]: Initially they were separate files :) There is some type error with the docker reuse that I could not solve. This is for now purely for speeding the integration tests of imports.

File: src/it/scala/com/exasol/cloudetl/DataImporterIT.scala [hunk 3]
```diff
@@ -0,0 +1,1108 @@
...
+        .assertResultSet(
+          table().row(java.lang.Byte.valueOf("11")).row(null).matches(NO_JAVA_TYPE_CHECK)
+        )
+    }
+
+    test("imports short") {
+      OrcChecker("struct<f:smallint>", "DECIMAL(9,0)")
+        .withInputValues(List(13, null))
+        .assertResultSet(
+          table().row(java.lang.Short.valueOf("13")).row(null).matches(NO_JAVA_TYPE_CHECK)
```
[the remark this answers is not in the document]
Comment (morazow) [reply src/it/scala/com/exasol/cloudetl/DataImporterIT.scala:402]: It is other way around. Even though we import short type, the Exasol table column type is INT (or decimal(9,0)). We want to assert it is short in the JDBC resultset.

[the remark this answers is not in the document]
Comment (morazow) [reply src/it/scala/com/exasol/cloudetl/DataImporterIT.scala:402]: The "13" by itself becomes int by default.

File: src/it/scala/com/exasol/cloudetl/BaseIntegrationTest.scala [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    exasolContainer.start()
+    s3Container.start()
+  }
+
+  def prepareExasolDatabase(schemaName: String): Unit = {
+    executeStmt(s"DROP SCHEMA IF EXISTS $schemaName CASCADE;")
+    factory = new ExasolObjectFactory(getConnection())
+    schema = factory.createSchema(schemaName)
+    createDeploymentScripts()
+    createConnectionObject()
```
[the remark this answers is not in the document]
Comment (morazow) [reply src/it/scala/com/exasol/cloudetl/BaseIntegrationTest.scala:44]: Fixed.

File: src/it/scala/com/exasol/cloudetl/DataImporterIT.scala [hunk 5]
```diff
@@ -0,0 +1,1108 @@
...
+        .withInputValues(List(true, false))
+        .assertResultSet(
+          table()
+            .row(java.lang.Boolean.TRUE)
+            .row(java.lang.Boolean.FALSE)
+            .matches()
+        )
+    }
+
+    test("imports int") {
```
[the remark this answers is not in the document]
Comment (morazow) [reply src/it/scala/com/exasol/cloudetl/DataImporterIT.scala:94]: Fixed.
Added tests for min&max values checks of integers and longs.

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**